src/controllers/FacebookController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Facebook-specific endpoints live here
+ */
+
+use GuzzleHttp\Client;
+
+class FacebookController extends ApiController
+{
+    /**
+     * Take the verification code from the client, send to Facebook to get an access token.
+     * With the access token, read the user's profile to get their email address.
+     * From that, look up who they are, create them a new access token and return
+     * the same format as we do when logging in a user
+     */
+    public function logUserIn($request, $db)
```

Comment:
It might be good to gate this function with a config parameter, to allow its disabling; there are config fields that are required further down that there's no sanity check on, and it will probably be handy to disable this functionality when an installation does not have an associated Facebook app (I'm assuming we're intending Joind.in to be portable)